libcxx/include/any

```diff
@@ -196,9 +196,7 @@ namespace __any_imp
       if (__id && *__id == typeid(_Tp))
           return true;
 #endif
-      if (!__id && __fallback_id == __any_imp::__get_fallback_typeid<_Tp>())
-          return true;
-      return false;
+      return !__id && __fallback_id == __any_imp::__get_fallback_typeid<_Tp>();
```

Comment:
Is there a reason why this is part of this patch?

Reply:
Yes the removal of the `#if !defined(_LIBCPP_HAS_NO_RTTI)` block gives a clang-tidy warning.